Library/Homebrew/cask/artifact/shim_script.rb

```diff
@@ -0,0 +1,195 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "cask/artifact/relocated"
+
+module Cask
+  module Artifact
+    # Artifact corresponding to a `shim_script` stanza.
+    #
+    # Purpose: many casks currently create small wrapper scripts in preflight
+    # blocks. This artifact standardizes that behavior.
+    #
+    # Usage examples in a cask:
+    #   shim_script "App.app/Contents/MacOS/foo", target: "foo"
+    #   shim_script "bin/tool", target: "tool", args: ["--flag"], env: { "FOO" => "bar" }
+    #   shim_script "App.app/Contents/MacOS/foo", target: "foo" do |source|
+    #     "#!/bin/sh\nexec \"#{source}\" --special \"$@\"\n"
+    #   end
```

Comment:
I'm not a big fan of this API (and maybe DSL) as it feels like it's just abstracting a lot of things that we actually care about.

Supporting something like `Formula#write_jar_script` or `Pathame#write_env_script` or `Pathname#env_script_all_files` with a DSL feels like it might make sense but, as-is, this feels like it's a bit overly magic wrapper around generating some Bash.